### PR TITLE
Move the xray allowdot feature gate to beta stage

### PR DESCRIPTION
**Pull request by XinRanZhAWS**
**Description:** 
Moving xray exporter allow-dot feature gate from alpha to beta stage

It's a follow up change for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31732

**Testing:** Unit test

**Documentation:** 

https://github.com/XinRanZhAWS/opentelemetry-collector-contrib/blob/f0863f481445e7205e3096a9a536463e3a51de81/.chloggen/xray_exporter_allow_dot_annotation.yaml

Purpose of this feature gate:

In the past, X-Ray doesn’t support “.”. So we have a translator in x-ray export to translates it to “_” before sending traces to X-Ray Service.

To match otel naming style, x-ray service team decide to change their service to support both "." type and "" type of naming. In this case the translator that translate "." to "" is no-longer needed. This PR change the way this translator work

X-Ray PMs agree on rolling out this change by using feature-gate

### CODE REVIEW

**Review by crobert-1 — COMMENTED** [review]
This needs a changelog that properly reflects this is a breaking change, since the feature gate will now be enabled by default.

**Comment by XinRanZhAWS** [comment]
> This needs a changelog that properly reflects this is a breaking change, since the feature gate will now be enabled by default.

Thanks for your comment, added changelog. Let me know if any else concern

**Review by songy23 — APPROVED** [review]

File: .chloggen/xrayexporter-dot-converter-featuregate-beta.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awsxrayexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: change x-ray exporter's translator to make "." split annotation pass as-is
```
Comment (crobert-1) [inline .chloggen/xrayexporter-dot-converter-featuregate-beta.yaml:10]: This should mention the feature gate name, and the fact that it's being upgraded from `Alpha` to `Beta`, along with the description you already have here.

Comment (XinRanZhAWS) [reply .chloggen/xrayexporter-dot-converter-featuregate-beta.yaml:10]: Done, thanks

**Review by mxiamxia — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**